### PR TITLE
added main routing, data mocks, styles reset

### PR DESCRIPTION
**Pull request by Ponomaleks**

### CODE REVIEW

File: src/app/main/components/chat-list-item/chat-list-item.component.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div style="border: 1px solid black" (click)="openChat()">
```
Comment (GennadiyShpak) [inline src/app/main/components/chat-list-item/chat-list-item.component.html:1]: remove inline styles please

File: src/app/main/components/chat-list-item/chats-element.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@Component({
+  selector: 'epm-chat-list-item',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './chat-list-item.component.html',
+  styleUrls: ['./chat-list-item.component.scss']
+})
+export class ChatsElementComponent {
+  @Input() chat: any;
```
Comment (GennadiyShpak) [inline src/app/main/components/chat-list-item/chats-element.component.ts:12]: why any?

File: src/app/main/components/contact-list-item/contact-list-item.component.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<div style="border: 1px solid black" (click)="openUser()">
```
Comment (GennadiyShpak) [inline src/app/main/components/contact-list-item/contact-list-item.component.html:1]: inline styles

File: src/app/main/main.model.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  about: string;
+  interests?: string[];
+  city: string;
+  avatarUrl?: string;
+  instagram?: string | null;
+  facebook?: string | null;
+  Skype?: string | null;
+  Teams?: string | null;
+  Telegram?: string | null;
+}
```
Comment (GennadiyShpak) [inline src/app/main/main.model.ts:23]: many fields are repeat

File: src/app/main/pages/chat-list/chat-list.component.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<p>chat-list works!</p>
+<ul>
+  <li *ngFor="let chat of chats">
```
Comment (GennadiyShpak) [inline src/app/main/pages/chat-list/chat-list.component.html:3]: don`t forget about trackBy

File: src/app/main/pages/contact-list/contact-list.component.html [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<p>contact-list works!</p>
+<ul>
+  <li *ngFor="let user of users">
```
Comment (GennadiyShpak) [inline src/app/main/pages/contact-list/contact-list.component.html:3]: trackBy

File: src/app/main/pages/contact-list/contact-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  selector: 'epm-contact-list',
+  standalone: true,
+  imports: [CommonModule, ChatsElementComponent, ContactListItemComponent],
+  templateUrl: './contact-list.component.html',
+  styleUrls: ['./contact-list.component.scss']
+})
+export class ContactListComponent {
+  public users: User[] = USERS;
+  constructor(private router: Router) {}
+  onOpenUser(userId: number) {
```
Comment (GennadiyShpak) [inline src/app/main/pages/contact-list/contact-list.component.ts:20]: spaces between rows

File: src/app/main/pages/chat-list/chat-list.component.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  imports: [CommonModule, ChatsElementComponent],
+  templateUrl: './chat-list.component.html',
+  styleUrls: ['./chat-list.component.scss']
+})
+export class ChatListComponent {
+  public chats: Chat[] = CHATS;
+  constructor(private router: Router) {}
+  onOpenChat(userId: number) {
+    if (userId) {
+      this.router.navigateByUrl(`${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.Chat}/${userId}`).then();
```
Comment (HellraiserCW) [inline src/app/main/pages/chat-list/chat-list.component.ts:21]: Gena advised to ignore .then()

File: src/app/main/pages/contact-list/contact-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    if (userId) {
+      this.router.navigateByUrl(`${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.Contact}/${userId}`).then();
```
Comment (HellraiserCW) [inline src/app/main/pages/contact-list/contact-list.component.ts:22]: And here ignore .then()

File: src/app/main/components/chat-list-item/chat-list-item.component.html (hunk 1 above)
Comment (Ponomaleks) [reply src/app/main/components/chat-list-item/chat-list-item.component.html]: removed

File: src/app/main/components/chat-list-item/chats-element.component.ts (hunk 2 above)
Comment (Ponomaleks) [reply src/app/main/components/chat-list-item/chats-element.component.ts]: fixed

File: src/app/main/components/contact-list-item/contact-list-item.component.html (hunk 3 above)
Comment (Ponomaleks) [reply src/app/main/components/contact-list-item/contact-list-item.component.html]: removed

File: src/app/main/main.model.ts (hunk 4 above)
Comment (Ponomaleks) [reply src/app/main/main.model.ts]: Fixed

File: src/app/main/pages/chat-list/chat-list.component.html (hunk 5 above)
Comment (Ponomaleks) [reply src/app/main/pages/chat-list/chat-list.component.html]: Thanks, done

File: src/app/main/pages/contact-list/contact-list.component.html (hunk 6 above)
Comment (Ponomaleks) [reply src/app/main/pages/contact-list/contact-list.component.html]: added

File: src/app/main/pages/contact-list/contact-list.component.ts (hunk 7 above)
Comment (Ponomaleks) [reply src/app/main/pages/contact-list/contact-list.component.ts]: added

File: src/app/main/pages/chat-list/chat-list.component.ts (hunk 8 above)
Comment (Ponomaleks) [reply src/app/main/pages/chat-list/chat-list.component.ts]: fixed

File: src/app/main/pages/contact-list/contact-list.component.ts (hunk 9 above)
Comment (Ponomaleks) [reply src/app/main/pages/contact-list/contact-list.component.ts]: fixed

File: src/app/app.routes.ts [hunk 10]
```diff
@@ -9,11 +9,11 @@ export const APP_ROUTES: Routes = [
   },
   {
     path: APP_ROUTER_NAME.Auth,
-    loadChildren: () => import('./auth/container/auth.routes')
+    loadChildren: () => import('./auth/container/auth.routes').then()
```
Comment (GennadiyShpak) [inline src/app/app.routes.ts:12]: wtf?

File: src/app/app.routes.ts [hunk 11]
```diff
@@ -9,11 +9,11 @@ export const APP_ROUTES: Routes = [
...
   },
   {
     path: APP_ROUTER_NAME.Main,
-    loadChildren: () => import('./main/container/main.routes')
+    loadChildren: () => import('./main/container/main.routes').then()
```
Comment (GennadiyShpak) [inline src/app/app.routes.ts:16]: and here?

File: src/app/main/components/chat-list-item/chat-list-item.component.html [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<div class="avatar">
+  <div class="avatar-wrapper">
+    <img [src]="chat.avatarUrl" alt="" />
```
Comment (GennadiyShpak) [inline src/app/main/components/chat-list-item/chat-list-item.component.html:3]: empty alt

File: src/app/main/components/chat-list-item/chat-list-item.component.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+  imports: [CommonModule],
+  templateUrl: './chat-list-item.component.html',
+  styleUrls: ['./chat-list-item.component.scss']
+})
+export class ChatListItemComponent {
+  @Input() chat!: ChatLastMessage;
+  @Output() openChat: EventEmitter<string> = new EventEmitter<string>();
+  @HostListener('click') onOpenChat(): void {
+    this.openChat.emit(this.chat.userId);
+  }
```
Comment (GennadiyShpak) [inline src/app/main/components/chat-list-item/chat-list-item.component.ts:17]: space between rows

Comment (GennadiyShpak) [inline src/app/main/components/chat-list-item/chat-list-item.component.ts:17]: Do we really need this hostlistener?

File: src/app/main/container/main.routes.ts [hunk 14]
```diff
@@ -1,8 +1,45 @@
...
+        component: ChatListComponent
+      },
+      {
+        path: `${APP_ROUTER_NAME.Chat}/:userId`,
+        component: ChatComponent
+      },
+      {
+        path: APP_ROUTER_NAME.MyProfile,
+        loadComponent: () => import('../pages/profile/profile.component'),
+        data: 'My Profile'
```
Comment (GennadiyShpak) [inline src/app/main/container/main.routes.ts:37]: why do we need it?

File: src/app/main/pages/chat-list/chat-list.component.ts [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(private router: Router) {}
+
+  onOpenChat(userId: string) {
+    if (userId) {
+      this.router.navigateByUrl(`${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.Chat}/${userId}`);
+    }
+  }
+
+  trackByUserId(_index: number, chat: ChatLastMessage): string {
+    return chat?.userId;
```
Comment (GennadiyShpak) [inline src/app/main/pages/chat-list/chat-list.component.ts:28]: can we avoid to use optional chaining?

File: src/app/main/components/contact-list-item/contact-list-item.component.ts [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+  templateUrl: './contact-list-item.component.html',
+  styleUrls: ['./contact-list-item.component.scss']
+})
+export class ContactListItemComponent {
+  @Input() user!: UserDetailed;
+  @Output() openUser: EventEmitter<string> = new EventEmitter<string>();
+
+  onOpenUser(): void {
+    this.openUser.emit(this.user.userId);
+  }
```
Comment (GennadiyShpak) [inline src/app/main/components/contact-list-item/contact-list-item.component.ts:18]: do we need it?

File: src/app/main/pages/contact-list/contact-list.component.ts [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+
+  constructor(private router: Router) {}
+
+  onOpenUser(userId: string) {
+    if (userId) {
+      this.router.navigateByUrl(`${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.Contact}/${userId}`);
+    }
+  }
+  trackByUserId(_index: number, user: UserDetailed): string {
+    return user?.userId;
```
Comment (GennadiyShpak) [inline src/app/main/pages/contact-list/contact-list.component.ts:28]: Optional chaining?

File: src/app/app.routes.ts (hunk 10 above)
Comment (Ponomaleks) [reply src/app/app.routes.ts:12]: thanks for the solution

File: src/app/app.routes.ts (hunk 11 above)
Comment (Ponomaleks) [reply src/app/app.routes.ts:16]: fixed

File: src/app/main/components/chat-list-item/chat-list-item.component.html (hunk 12 above)
Comment (Ponomaleks) [reply src/app/main/components/chat-list-item/chat-list-item.component.html:3]: alt added

File: src/app/main/components/chat-list-item/chat-list-item.component.ts (hunk 13 above)
Comment (Ponomaleks) [reply src/app/main/components/chat-list-item/chat-list-item.component.ts]: Moved click logic to the list level

File: src/app/main/container/main.routes.ts (hunk 14 above)
Comment (Ponomaleks) [reply src/app/main/container/main.routes.ts]: It was an extra. Removed

File: src/app/main/pages/chat-list/chat-list.component.ts (hunk 15 above)
Comment (Ponomaleks) [reply src/app/main/pages/chat-list/chat-list.component.ts]: removed

File: src/app/main/components/contact-list-item/contact-list-item.component.ts (hunk 16 above)
Comment (Ponomaleks) [reply src/app/main/components/contact-list-item/contact-list-item.component.ts]: Moved click logic to the list level

File: src/app/main/pages/contact-list/contact-list.component.ts (hunk 17 above)
Comment (Ponomaleks) [reply src/app/main/pages/contact-list/contact-list.component.ts]: removed

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**